pysqldb3/pysqldb3.py

```diff
@@ -2073,3 +2074,216 @@ def feature_class_to_table(self, path, table, schema=None, shp_name=None, gdal_d
+            FROM
+            INFORMATION_SCHEMA.COLUMNS
+            WHERE TABLE_NAME = '{}'
+            """.format(tmp_table_name), internal=True)
+
+            cols = ['\\"' + c + '\\"' for c in list(col_df['column_name'])]
+            dt_col_names = ['\\"' + c + '\\"' for c in list(
+                col_df[col_df['data_type'].str.contains('datetime') | col_df['data_type'].str.contains('timestamp')][
+                    'column_name'])]
+
+        elif self.type == MS:
+            col_df = self.dfquery(f"""
+            SELECT
+                [column] = c.name,
+                [type] = t.name, 
+                c.max_length, 
+                c.precision, 
+                c.scale, 
+                c.is_nullable
+            FROM
+                tempdb.sys.columns AS c
+            LEFT JOIN
+                tempdb.sys.types AS t
+            ON
+                c.system_type_id = t.system_type_id
+                AND
+                t.system_type_id = t.user_type_id
+            WHERE
+                [object_id] = OBJECT_ID(N'tempdb.dbo.#{tmp_table_name}');
+            """, internal=True)
+
+            cols = ['[' + c + ']' for c in list(col_df['column'])]
+            dt_col_names = ['[' + c + ']' for c in list(
+                col_df[col_df['type'].str.contains('datetime') | col_df['type'].str.contains('timestamp')]['column'])]
+
+        # Make string of columns to be returned by select statement
+        return_cols = ' , '.join([c for c in cols if c not in dt_col_names])
+
+        # If there are datetime/timestamp columns:
+        if len(dt_col_names) > 0:
+            if self.type == PG:
+                print_cols = str([str(c[2:-2]) for c in dt_col_names])
+
+            if self.type == MS:
+                print_cols = str([str(c[1:-1]) for c in dt_col_names])
+
+            # Add the date and time (casted as a string) to the output
+            for col_name in dt_col_names:
+                if self.type == PG:
+                    shortened_col = col_name[2:-2][:7]
+                    return_cols += ' , cast(\\"{col}\\" as date) \\"{short_col}_dt\\", ' \
+                                   'cast(cast(\\"{col}\\" as time) as varchar) \\"{short_col}_tm\\" '.format(
+                                    col=col_name[2:-2], short_col=shortened_col)
+                elif self.type == MS:
+                    shortened_col = col_name[1:-1][:7]
+                    return_cols += " , cast([{col}] as date) [{short_col}_dt], cast(cast([{col}] as time) as varchar)" \
+                                   " [{short_col}_tm] ".format(
+                                    col=col_name[1:-1], short_col=shortened_col)
+
+        # Wrap the original query and select the non-datetime/timestamp columns and the parsed out dates/times
+        new_query = u"select {} from ({}) q ".format(return_cols, query)
+        Query.query_to_gpkg(self, new_query, path=path, gpkg_name=gpkg_name, cmd=cmd, gdal_data_loc=gdal_data_loc,
```

Comment:
everything besides this call is the same a qry to shp, so you should extract the code that is the same into a sub/internal function that both can call and just pass in the call to different the Query methods

Reply:
ok i will change this